receiver/mongodbreceiver/metrics.go

```diff
@@ -269,6 +307,147 @@ func (s *mongodbScraper) recordGlobalLockTime(now pcommon.Timestamp, doc bson.M,
+}
+
+func (s *mongodbScraper) recordNetworkCount(now pcommon.Timestamp, doc bson.M, errors scrapererror.ScrapeErrors) {
+	// Collect network bytes receive, transmit, and request count
+	networkRecorderMap := map[string]func(pcommon.Timestamp, int64){
+		"bytesIn":     s.mb.RecordMongodbNetworkIoReceiveDataPoint,
+		"bytesOut":    s.mb.RecordMongodbNetworkIoTransmitDataPoint,
+		"numRequests": s.mb.RecordMongodbNetworkRequestCountDataPoint,
+	}
+	for networkKey, recorder := range networkRecorderMap {
+		network, err := dig(doc, []string{"network", networkKey})
+		if err != nil {
+			errors.AddPartial(1, err)
+			s.logger.Error("failed to find network", zap.Error(err), zap.String("network", networkKey))
+		}
+		networkValue, err := parseInt(network)
+		if err != nil {
+			errors.AddPartial(1, err)
+		} else {
+			recorder(now, networkValue)
+		}
+	}
+}
+
+// Index Stats
+func (s *mongodbScraper) recordIndexAccess(now pcommon.Timestamp, documents []bson.M, dbName string, collectionName string, scraperErrors scrapererror.ScrapeErrors) {
+	// Collect the index access given a collection and database
+	var indexAccessTotal int64
+	for _, doc := range documents {
+		indexAccess, ok := doc["accesses"].(bson.M)["ops"]
+		if !ok {
+			scraperErrors.AddPartial(1, errors.New("could not find key for metric"))
+			return
+		}
+		indexAccessValue, err := parseInt(indexAccess)
+		if err != nil {
+			scraperErrors.AddPartial(1, err)
+			return
+		}
+		indexAccessTotal += indexAccessValue
+	}
+	s.mb.RecordMongodbIndexAccessCountDataPoint(now, indexAccessTotal, dbName, collectionName)
+}
+
+// Top Stats
+func (s *mongodbScraper) recordOperationTime(now pcommon.Timestamp, doc bson.M, scraperErrors scrapererror.ScrapeErrors) {
+	// Collect the total operation time
+	collectionPathNames, err := digForCollectionPathNames(doc)
+	if err != nil {
+		scraperErrors.AddPartial(1, err)
+		return
+	}
+	operationTimeValues, err := aggregateOperationTimeValues(doc, collectionPathNames)
+	if err != nil {
+		scraperErrors.AddPartial(1, err)
+		return
+	}
+
+	operationsMap := map[string]metadata.AttributeOperation{
+		"insert":   metadata.AttributeOperationInsert,
+		"queries":  metadata.AttributeOperationQuery,
+		"update":   metadata.AttributeOperationUpdate,
+		"remove":   metadata.AttributeOperationDelete,
+		"getmore":  metadata.AttributeOperationGetmore,
+		"commands": metadata.AttributeOperationCommand,
+	}
+	for operationName, metadataOperationName := range operationsMap {
+		operationValue, ok := operationTimeValues[operationName]
+		if !ok {
+			scraperErrors.AddPartial(1, errors.New("could not find key for metric"))
+			return
+		}
+		s.mb.RecordMongodbOperationTimeDataPoint(now, operationValue, metadataOperationName)
+	}
+}
+
+func aggregateOperationTimeValues(document bson.M, collectionPathNames []string) (map[string]int64, error) {
+	operationTotals := map[string]int64{}
+	operationNames := []string{"insert", "queries", "update", "remove", "getmore", "commands"}
```

Comment:
could be useful to pull constants out and put at the top of the file for easier debugging/updating

Reply:
Updated. Moved operation names / mappings outside of functions.